spring-integration-graphql/src/test/java/org/springframework/integration/graphql/outbound/GraphQlMessageHandlerTests.java

```diff
@@ -0,0 +1,424 @@
+		@MutationMapping
+		public Mono<Update> update(@Argument String id) {
+			return this.updateRepository.save(new Update(id));
+		}
+
+		@SubscriptionMapping
+		public Flux<QueryResult> results() {
+			return Flux.just(
+					new QueryResult("test-data-01"),
+					new QueryResult("test-data-02"),
+					new QueryResult("test-data-03"),
+					new QueryResult("test-data-04"),
+					new QueryResult("test-data-05"),
+					new QueryResult("test-data-06"),
+					new QueryResult("test-data-07"),
+					new QueryResult("test-data-08"),
+					new QueryResult("test-data-09"),
+					new QueryResult("test-data-10")
+			);
+		}
+
+	}
+
+	@Repository
+	static class UpdateRepository {
+
+		private Update current;
+
+		Mono<Update> save(Update update) {
+			this.current = update;
+			return Mono.justOrEmpty(this.current);
+		}
+
+		Mono<Update> current() {
+			return Mono.just(this.current);
+		}
+
+	}
+
+	@Configuration
+	@EnableIntegration
+	static class TestConfig {
+
+		@Bean
+		GraphQlMessageHandler handler(GraphQlService graphQlService) {
+
+			return new GraphQlMessageHandler(graphQlService);
+		}
+
+		// @artem
+		@Bean
+		IntegrationFlow graphqlQueryMessageHandlerFlow(GraphQlMessageHandler handler) {
+
+			return IntegrationFlows.from(MessageChannels.flux("inputChannel").datatype(Object.class, RequestInput.class))
+					.handle(handler)
+					.channel(c -> c.flux("resultChannel").datatype(RequestOutput.class))
+					.get();
+		}
+
+		@Bean
+		PollableChannel errorChannel() {
+
+			return new QueueChannel();
+		}
+
+		@Bean
+		UpdateRepository updateRepository() {
+			return new UpdateRepository();
+		}
+
+		@Bean
+		GraphQlController graphqlQueryController(final UpdateRepository updateRepository) {
+
+			return new GraphQlController(updateRepository);
+		}
+
+		@Bean
+		GraphQlService graphQlService(GraphQlSource graphQlSource) {
+
+			return new ExecutionGraphQlService(graphQlSource);
+		}
+
+		@Bean
+		GraphQlSource graphQlSource(AnnotatedControllerConfigurer annotatedDataFetcherConfigurer) {
+
+			return GraphQlSource.builder()
+					.schemaResources(new ClassPathResource("graphql/test-schema.graphqls"))
+					.configureRuntimeWiring(annotatedDataFetcherConfigurer)
+					.build();
+		}
+
+		@Bean
+		AnnotatedControllerConfigurer annotatedDataFetcherConfigurer() {
+
+			return new AnnotatedControllerConfigurer();
+		}
+
+	}
+
+	static class QueryResult {
+
+		private final String id;
+
+		QueryResult(final String id) {
+			this.id = id;
+		}
+
+		String getId() {
+			return this.id;
+		}
+
+		@Override
+		public boolean equals(Object o) {
+			if (this == o) {
+				return true;
+			}
+			if (!(o instanceof QueryResult)) {
+				return false;
+			}
+			QueryResult that = (QueryResult) o;
+			return getId().equals(that.getId());
+		}
+
+		@Override
+		public int hashCode() {
+			return Objects.hash(getId());
+		}
+
+		@Override
+		public String toString() {
+			return "QueryResult{" +
+					"id='" + id + '\'' +
+					'}';
+		}
+	}
+
+	static class Update {
```

Comment:
These two last classes could be as simple as:
```
	record QueryResult(String id) {	}

	record Update(String id) { 	}
```
No need in that extra boilerplate code 😄 !